node/worker/worker.go

```diff
@@ -85,6 +87,14 @@ func (w *Worker) throttleTxs(selected types.Transactions, recentTxsStats types.R
 
 // SelectTransactionsForNewBlock selects transactions for new block.
 func (w *Worker) SelectTransactionsForNewBlock(newBlockNum uint64, txs types.Transactions, recentTxsStats types.RecentTxsStats, txsThrottleConfig *shardingconfig.TxsThrottleConfig, coinbase common.Address) (types.Transactions, types.Transactions, types.Transactions) {
+	// Must update to the correct current state before processing potential txns
+	if err := w.UpdateCurrent(coinbase); err != nil {
```

Comment:
updatecurrent is called in here and commit transactions, can we remove one of them?

Reply:
We need both of them.